src/token_cache.rs

```diff
@@ -1,17 +1,19 @@
 use crate::reference::Reference;
 use serde::Deserialize;
 use std::collections::BTreeMap;
+use std::time::{SystemTime, UNIX_EPOCH};
 use tracing::{debug, warn};
 
 /// A token granted during the OAuth2-like workflow for OCI registries.
-#[derive(Deserialize, Clone)]
+#[derive(Deserialize, Debug, Clone)]
 #[serde(untagged)]
 #[serde(rename_all = "snake_case")]
 pub(crate) enum RegistryToken {
     Token { token: String },
     AccessToken { access_token: String },
 }
 
+#[derive(Debug)]
```

Comment:
I worry about deriving debug here as it could leak sensitive data when used in a `println!`. If we need a debug, we should manually implement it so as to mask any sensitive data

Reply:
Good point, I've fixed that